src/theory/quantifiers/ematching/inst_strategy_e_matching.cpp

```diff
@@ -360,7 +360,16 @@ void InstStrategyAutoGenTriggers::generateTriggers( Node f ){
     if( d_num_trigger_vars[f]>0 && d_num_trigger_vars[f]<f[0].getNumChildren() ){
       Trace("auto-gen-trigger-partial") << "Quantified formula : " << f << std::endl;
       Trace("auto-gen-trigger-partial") << "...does not contain all variables in triggers!!!" << std::endl;
-      if( options::partialTriggers() ){
+      // Invoke partial trigger strategy: partition variables of quantified
+      // formula into (X,Y) where X are contained in a trigger and Y are not.
+      // We then force a split of the quantified formula so that it becomes:
+      //   forall X. forall Y. P( X, Y )
+      // and hence is treatable by E-matching. We only do this for "standard"
+      // quantified formulas (those with only two children), since this
+      // technique should not be used for e.g. quantifiers marked for
+      // quantifier elimination.
+      if (options::partialTriggers() && f.getNumChildren() == 2)
```

Comment:
It'd be nice if you could abstract this test of being an "standard quantifier", what do you think? It might improve the code's legibility.

Reply:
Actually this has been done already: https://github.com/CVC4/CVC4/blob/master/src/theory/quantifiers/quantifiers_attributes.h#L157
although I need to double check what cases a quantified formula can be standard while having 3 children.  (This code was being overly cautious/simple).

Reply:
I addressed this in my latest commits.

Reply:
Great, thanks!